src/main/scala/firrtl/Emitter.scala

```diff
@@ -412,7 +413,12 @@ class VerilogEmitter extends SeqTransform with Emitter {
     val assigns = ArrayBuffer[Seq[Any]]()
     val attachSynAssigns = ArrayBuffer.empty[Seq[Any]]
     val attachAliases = ArrayBuffer.empty[Seq[Any]]
-    val at_clock = mutable.LinkedHashMap[Expression, ArrayBuffer[Seq[Any]]]()
+    // No (aka synchronous) always blocks, keyed by clock
+    val noResetAlwaysBlocks = mutable.LinkedHashMap[Expression, ArrayBuffer[Seq[Any]]]()
+    // One always block per async reset register, (Clock, Reset, Content)
+    // An alternative approach is to have one always block per combination of clock and async reset,
+    // but Formality doesn't allow more than 1 statement inside async reset always blocks
+    val asyncResetAlwaysBlocks = mutable.ArrayBuffer[(Expression, Expression, Seq[Any])]()
```

Comment:
a: "Chisel emits unreadable, verbose Verilog."
b: "Synopsys *demands* unreadable, verbose Verilog."

???

:hankey: :hankey: :hankey:

Reply:
Profit?

Reply:
We can now sell the hard to read verilog as tool-friendly and kill two birds with one stone.

Reply:
Put this man on the product marketing team

Reply:
But seriously this may actually be a good idea. There's a lot of effort in the Verilog world of top-down brittle coding styles because they produce better results [[1], [2], [3], [4]]. If these are encapsulated in either the FIRRTL optimizations or in the Verilog emitter then you don't wind up with brittle "code it this way because our Cadence rep told us to" anti-patterns. That also may help avoid vendor lock in.

At some point, it would likely be sensible to encapsulate these types of "optimizations" in something like an `-mtool` flag, e.g., `-mtool=synopsys` vs. `-mtool=cadence`. 

[1]: http://www.sunburst-design.com/papers/CummingsSNUG2000SJ_NBA.pdf
[2]: http://www.sunburst-design.com/papers/CummingsSNUG1999SJ_SynthMismatch.pdf
[3]: http://www.sunburst-design.com/papers/CummingsSNUG1998SJ_FSM.pdf
[4]: http://www.sunburst-design.com/papers/CummingsICU1997_VerilogCodingEfficiency.pdf

Reply:
Even better is to allow downstream users to customize the emitter as needed. e.g. I may want to ship a emitter-modifying pass with Hammer that makes adjustments depending on what CAD tools are selected.

Reply:
We obviously want to keep the Verilog as compatible as possible but it's abundantly clear that different styles are sometimes necessary for different tools. Customization of emission along with generated collateral to be confident that the emitted code is the same (ie. LEC or model checking scripts) seems like a reasonable approach to me.